### PR TITLE
feat: publish snapshot on merge to main

### PR DESCRIPTION
**Pull request by bnpfeife**
This builds and pushes artifacts to the snapshot bucket on merge to main. The snapshot workflow does not include the tests run from within the PR as it would duplicate work and increase CI costs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -278,21 +278,24 @@ jobs:
     parameters:
       target:
         type: string
+      profile:
+        type: string
+        default: release
```
Comment (hiltontj) [inline .circleci/config.yml:283]: Did you want to change this to `quick-release` by default to reduce CI costs for the immediate term?

**Review by hiltontj — APPROVED** [review]
See one comment, otherwise LGTM, thanks @bnpfeife

File: .circleci/config.yml (hunk 1 above)
Comment (bnpfeife) [reply .circleci/config.yml:283]: Yup, it's set to quick-release by the workflow. It's on line 489.
